### PR TITLE
app: Logout with active orders error

### PR DESCRIPTION
**Pull request by martonp**
Previously if the user tried to log out while there were active deals, there would be no feedback. Now, an error message is displayed.

Screen Shot 2022-10-16 at 7 54 11 AM

Closes: #1895

### CODE REVIEW

File: client/webserver/site/src/js/wallets.ts [hunk 1]
```diff
@@ -962,7 +962,11 @@ export default class WalletsPage extends BasePage {
     const loaded = app().loading(page.newWalletForm)
     const res = await postJSON('/api/closewallet', { assetID: assetID })
     loaded()
-    if (!app().checkResponse(res)) return
+    if (!app().checkResponse(res)) {
+      page.walletErr.textContent = `Error locking wallet: ${res.msg}`
```
Comment (ukane-philemon) [inline client/webserver/site/src/js/wallets.ts:966]: might need to add this hardcoded string to translation at `locales.ts`.

File: client/webserver/site/src/js/app.ts [hunk 2]
```diff
@@ -887,9 +888,13 @@ export default class Application {
   async signOut () {
     const res = await postJSON('/api/logout')
     if (!this.checkResponse(res)) {
-      // TODO: Esp. for active orders error, let the user know what's going on.
-      // Maybe give them an option to force.
-      Doc.hide(this.page.profileBox)
+      const activeOrdersErrCode = 35
```
Comment (chappjc) [inline client/webserver/site/src/js/app.ts:891]: We have this defined at https://github.com/decred/dcrdex/blob/34732a8b7e52346d86e52a03610845e450125059/client/webserver/site/src/js/wallets.ts#L37

Seems like we should have all http API error codes that we need defined somewhere central.  We had a constants.js at one point, but that got nuked.  Thoughts?  Maybe in http.ts or at the top of app.ts?

File: client/core/core.go [hunk 3]
```diff
@@ -4126,7 +4126,7 @@ func (c *Core) Logout() error {
 	conns := c.dexConnections()
 	for _, dc := range conns {
 		if dc.hasActiveOrders() {
-			return ActiveOrdersLogoutErr
+			return newError(activeOrdersErr, "%w", ActiveOrdersLogoutErr)
```
Comment (chappjc) [inline client/core/core.go:4129]: Without a message, just `codedError(activeOrdersErr, ActiveOrdersLogoutErr)`

**Review by chappjc — CHANGES_REQUESTED** [review]
Working as expected.  Just two minor changes please.

File: client/webserver/site/src/js/app.ts (hunk 2 above)
Comment (martonp) [reply client/webserver/site/src/js/app.ts]: I think that's a good idea. I put them in http.ts.

**Pull request merged**